llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -0,0 +1,534 @@
+        client = _default_retriever
+    try:
+        document = client.get_document(
+            genai.GetDocumentRequest(
+                name=str(EntityName(corpus_id=corpus_id, document_id=document_id))
+            )
+        )
+        return Document.from_document(document)
+    except Exception:
+        return None
+
+
+def create_document(
+    *,
+    corpus_id: str,
+    document_id: Optional[str] = None,
+    display_name: Optional[str] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> Document:
+    if client is None:
+        client = _default_retriever
+    name: Optional[str]
+    if document_id is not None:
+        name = str(EntityName(corpus_id=corpus_id, document_id=document_id))
+    else:
+        name = None
+
+    new_display_name = display_name or f"Untitled {datetime.datetime.now()}"
+    new_metadatas = _convertToMetadata(metadata) if metadata else None
+
+    new_document = client.create_document(
+        genai.CreateDocumentRequest(
+            parent=str(EntityName(corpus_id=corpus_id)),
+            document=genai.Document(
+                name=name, display_name=new_display_name, custom_metadata=new_metadatas
+            ),
+        )
+    )
+
+    return Document.from_document(new_document)
+
+
+def delete_document(
+    *,
+    corpus_id: str,
+    document_id: str,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> None:
+    if client is None:
+        client = _default_retriever
+    client.delete_document(
+        genai.DeleteDocumentRequest(
+            name=str(EntityName(corpus_id=corpus_id, document_id=document_id)),
+            force=True,
+        )
+    )
+
+
+def batch_create_chunk(
+    *,
+    corpus_id: str,
+    document_id: str,
+    texts: List[str],
+    metadatas: Optional[List[Dict[str, Any]]] = None,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> List[genai.Chunk]:
+    if client is None:
+        client = _default_retriever
+    if metadatas is None:
+        metadatas = [{} for _ in texts]
+    if len(texts) != len(metadatas):
+        raise ValueError(
+            f"metadatas's length {len(metadatas)} and texts's length {len(texts)} are mismatched"
+        )
+
+    doc_name = str(EntityName(corpus_id=corpus_id, document_id=document_id))
+
+    # Temporary solution until b/307125429 is fixed.
+    created_chunks: List[genai.Chunk] = []
+    for text, metadata in zip(texts, metadatas):
+        created_chunk = client.create_chunk(
+            genai.CreateChunkRequest(
+                parent=doc_name,
+                chunk=genai.Chunk(
+                    data=genai.ChunkData(string_value=text),
+                    custom_metadata=_convertToMetadata(metadata),
+                ),
+            )
+        )
+    return created_chunks
+
+
+"""Restore this after b/307125429 is fixed.
+operation = client.batch_create_chunks(
+    genai.BatchCreateChunksRequest(
```

Comment:
Is this a generated method? If so, we'll need to make sure each batch is no more than 100 chunks.

Reply:
Restored to BatchCreateChunks.